packages/quantic/force-app/main/default/lwc/quanticStandaloneSearchBox/__tests__/quanticStandaloneSearchBox.test.js

```diff
@@ -0,0 +1,153 @@
+/* eslint-disable no-import-assign */
+import QuanticStandaloneSearchBox from 'c/quanticStandaloneSearchBox';
+// @ts-ignore
+import {createElement} from 'lwc';
+import * as mockHeadlessLoader from 'c/quanticHeadlessLoader';
+
+jest.mock('c/quanticHeadlessLoader');
+
+let isInitialized = false;
+
+const exampleEngine = {
+  id: 'engineId__standalone',
+};
+
+const functionsMocks = {
+  buildStandaloneSearchBox: jest.fn(() => ({
+    state: {},
+    subscribe: functionsMocks.subscribe,
+  })),
+  loadDependencies: jest.fn(
+    () =>
+      new Promise((resolve) => {
+        resolve();
+      })
+  ),
+  subscribe: jest.fn((cb) => {
+    cb();
+    return functionsMocks.unsubscribe;
+  }),
+  unsubscribe: jest.fn(() => {}),
+};
+
+const defaultOptions = {
+  engineId: exampleEngine.id,
+  placeholder: null,
+  withoutSubmitButton: false,
+  numberOfSuggestions: 7,
+  textarea: false,
+  disableRecentQueries: false,
+  keepFiltersOnSearch: false,
+};
+
+function createTestComponent(options = defaultOptions) {
+  prepareHeadlessState();
+
+  const element = createElement('c-quantic-standalone-search-box', {
+    is: QuanticStandaloneSearchBox,
+  });
+  for (const [key, value] of Object.entries(options)) {
+    element[key] = value;
+  }
+  document.body.appendChild(element);
+  return element;
+}
+
+function prepareHeadlessState() {
+  // @ts-ignore
+  mockHeadlessLoader.loadDependencies = functionsMocks.loadDependencies;
```

Comment:
why is this needed?

Reply:
otherwise I was getting this error: 

<img width="589" alt="image" src="https://github.com/user-attachments/assets/675d3bdb-009c-4dfd-bec1-1bb908753bd1">